### PR TITLE
Make new modules importable

### LINKED ISSUE #5: New modules can not be imported
Currently, the following fails in the context of the documentation, i.e., when `submodule_new` is a module that has been introduced by the patch:
```
>>> import a_test_module.submodule_new
ModuleNotFoundError: No module named 'a_test_module.submodule_new'
```

### PR DESCRIPTION
**Pull request by saraedum**
Fixes #5.

### CODE REVIEW

File: recursive_monkey_patch.py [hunk 1]
```diff
@@ -206,6 +208,14 @@ def monkey_patch(source, target, log_level=logging.WARNING, logger=None):
 
         >>> A.NestedNew.__module__
         'a_test_module_patch.submodule'
+
+    .. RUBRIC:: Testing that new modules behave like old modules
```
Comment (nthiery) [inline recursive_monkey_patch.py]: Maybe: "like preexisting modules in the target"?

File: recursive_monkey_patch.py [hunk 2]
```diff
@@ -206,6 +208,14 @@ def monkey_patch(source, target, log_level=logging.WARNING, logger=None):
...
+
+    The new modules can be imported (actually they are already imported)::
+
+        >>> import a_test_module.submodule_new
+        >>> from a_test_module.submodule_new import i
```
Comment (nthiery) [inline recursive_monkey_patch.py]: What about displaying the value of `a_test_module.submodule.__module__`?

Comment (saraedum) [reply recursive_monkey_patch.py]: Hm, `AttributeError: 'module' object has no attribute '__module__'`

**Comment by nthiery** [comment]
I haven't yet wrapped around my head to understand and double check the new logic.
Is it sill possible to import the module under it's original name?
Could this possibly break (or actually improve) things like pickling of instances of classes defined in the module?

**Comment by saraedum** [comment]
I did this to make pickling with factories work :)

**Comment by saraedum** [comment]
Let me try the original import and add a doctest for it.

File: recursive_monkey_patch.py (hunk 2 above)
Comment (nthiery) [reply recursive_monkey_patch.py]: Sorry, I meant `.__name__`

**Comment by nthiery** [comment]
:-)

Could the module name renaming confuse introspection? E.g. what happens upon
```
    sage: a_test_module.submodule_new??
    sage: a_test_module_patch.submodule_new??
```
and similar in classes/functions defined in `submodule_new`?

**Comment by saraedum** [comment]
My last commit contains an alternative that is probably better?

**Comment by saraedum** [comment]
Let me check introspection now.

**Comment by saraedum** [comment]
Here is the behavior with my latest changes:
```
sage: from recursive_monkey_patch import monkey_patch
sage: import a_test_module_patch
sage: import a_test_module
sage: monkey_patch(a_test_module_patch, a_test_module)
sage: a_test_module.submodule_new??
Type:            module
String form:     <module 'a_test_module.submodule_new' from 'a_test_module_patch/submodule_new.py'>
File:            ~/proj/sage/recursive-monkey-patch/a_test_module_patch/submodule_new.py
Source:         
class New:
    r"""
    A docstring
    """
    pass
Class docstring:
module(name[, doc])

Create a module object. The name must be a string; the optional doc
argument can have any type.
Init docstring:  x.__init__(...) initializes x; see help(type(x)) for signature
sage: from a_test_module.submodule_new import New
sage: New??
Source:
class New:
    r"""
    A docstring
    """
    pass
File:   ~/proj/sage/recursive-monkey-patch/a_test_module_patch/submodule_new.py
Type:   classobj
sage: import a_test_module_patch.submodule_new
sage: a_test_module_patch.submodule_new??
Type:            module
String form:     <module 'a_test_module.submodule_new' from 'a_test_module_patch/submodule_new.py'>
File:            ~/proj/sage/recursive-monkey-patch/a_test_module_patch/submodule_new.py
Source:         
class New:
    r"""
    A docstring
    """
    pass
Class docstring:
module(name[, doc])

Create a module object. The name must be a string; the optional doc
argument can have any type.
Init docstring:  x.__init__(...) initializes x; see help(type(x)) for signature
sage: from a_test_module_patch.submodule_new import New as New_
sage: New_??
Source:
class New:
    r"""
    A docstring
    """
    pass
File:   ~/proj/sage/recursive-monkey-patch/a_test_module_patch/submodule_new.py
Type:   classobj
```

**Comment by saraedum** [comment]
Looks reasonable to me.

**Comment by saraedum** [comment]
~~@nthiery Have you enabled travis for Pull Requests? Would be nice to see it checking this now :)~~

Great. Seems to work.

**Comment by nthiery** [comment]
I agree the behaviour sounds sane. I did not get the time to review the logic; but it's not yet critical software, so tests + putting it into production will be good enough. Let's get this in.
Thanks much for the contribution!

**Pull request merged**